### ISSUE TITLE
Code analysis regression: missing methods from ext trait and wrong type inference causing non-existing error

### ISSUE THREAD

**Issue by Riateche** [rust-lang/rust-analyzer#20954]
**rust-analyzer version**:  0.4.2664-standalone (2e2e3ebec9 2025-10-31)

**rustc version**:  rustc 1.88.0 (6b00bc388 2025-06-23)

**editor or extension**: VS Code, rust-lang.rust-analyzer pre-release, Version 0.4.2664

**code snippet to reproduce**:
```rust
use std::any::Any;

pub trait A: Any {}

pub trait B: A {
    fn b_func(&self) -> S {
        S { min: V(0) }
    }
}

impl<T: A + ?Sized> B for T {}

fn f_accepts_v(_v: V, _v2: V) {}

pub fn f(dyn_a: &mut dyn A) {
    let s = dyn_a.b_func();
    f_accepts_v(s.min(), s.min());
}

#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
pub struct V(pub u64);

pub struct S {
    pub min: V,
}

impl S {
    pub fn min(&self) -> V {
        self.min
    }
}
```
The first issue is that rust-analyzer doesn't see `b_func` (e.g. "go to definition" from `dyn_a.b_func()` call location doesn't work), so rust-analyzer doesn't know the proper type of `s` variable.

The second issue is that rust-analyzer uses the next call context `f_accepts_v(s.min(), s.min())` to incorrectly infer that `s: V` (in fact, `s: S`). It then issues an incorrect error "expected 1 argument, found 0":

<img width="407" height="208" alt="Image" src="https://github.com/user-attachments/assets/db7a05e4-4de3-4729-adc3-cd4d98dacbb9" />

<img width="507" height="194" alt="screenshot showing rust-analyzer error" src="https://github.com/user-attachments/assets/a1088ab3-13a9-4884-80f5-e3992ef7ae04" />

rustc compiles this code without warnings or errors. This code was also previously handled correctly by rust-analyzer.

**Comment by dfireBird** [comment]
I can't seem to replicate this issue both on stable and on nightly release. Can you mention if you have some vscode setting on rust-analyzer side. 

P.S: Both on latest rust: `rustc 1.90.0 (1159e78c4 2025-09-14)`

**Comment by ChayimFriedman2** [comment]
We know we have issues with latest r-a with previous rustc versions (although I don't know what version range is problematic), can you try with latest Rust?

**Comment by Riateche** [comment]
Yeah, updating Rust toolchain fixes it, thanks!
